students/Dustin_L/lesson09/test_mailroom4.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python3
+"""This is a test module that tests mailroom4.py"""
+
+import datetime
+import os
+import random
+import sys
+from io import StringIO
+
+import unittest
+from unittest import mock
+
+import mailroom4 as mr
+from mailroom4 import PROMPT_OPTS, Donor, DonorDatabase
+
+
+def redirect_stdout():
+    """Redirect stdout to the returned StringIO object.
+
+    Returns:
+        StringIO: Object containing redirected stdout.
+    """
+    redirect = StringIO()
+    sys.stdout = redirect
+
+    return redirect
+
+
+def reset_stdout():
+    """Reset stdout back to default"""
+    sys.stdout = sys.__stdout__
+
+
+class TestDonor(unittest.TestCase):
+    """Unit tests for the Donor class"""
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.test_name = 'Test Donor'
+        self.test_dons = [100, 500, 50.05]
+        self.test_total = sum(self.test_dons)
+        self.test_ave = self.test_total / len(self.test_dons)
+
+    def setUp(self):
+        self.donor = Donor(self.test_name, self.test_dons)
+
+    def tearDown(self):
+        pass
+
+    def test_get_donations(self):
+        """Test donations attribute"""
+        self.assertTrue(self.donor.donations == self.test_dons)
+
+    def test_set_donations(self):
+        """Test setting donations attribute"""
+        with self.assertRaises(AttributeError):
+            self.donor.donations = [100, 20]
+
+    def test_get_num_donations(self):
+        """Test num_donations attribute"""
+        self.assertTrue(self.donor.num_donations == len(self.test_dons))
```

Comment:
`self.assertEquals`  I believe would give you a better error message on failure and also makes more sense here